### ISSUE TITLE
The route returning <fs::NamedFile> does not support reverse proxy for nginx http2

### ISSUE THREAD

**Issue by Hentioe** [actix/actix-web#583]
I need to add `proxy_http_version 1.1;` to my nginx configuration, for example:

````
server {
        listen       443 ssl http2;
        server_name  www.your.domain;

        ssl on;

        location / {
                proxy_http_version 1.1;
                proxy_set_header Host $host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header REMOTE-HOST $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                add_header Strict-Transport-Security "max-age=31536000;" always;
                proxy_pass http://0.0.0.0:8080;
                access_log /var/log/nginx/acitx_app.log;
        }
}
````
Otherwise the file could not be obtained and got the `ERR_SPDY_PROTOCOL_ERROR` error (in Chrome)

**Comment by DoumanAsh** [comment]
I'm not sure of details how nginx work, but you cannot use HTTP2 without SSL so your server would need to have SSL before it could communicate with nginx over HTTP2.

I think it is expected that you'd set http version, as using SSL between proxy and actix-web server makes a little sense

**Comment by Hentioe** [comment]
@DoumanAsh Thank you for your reply, I added some information.

I deployed a project to reproduce this phenomenon.

1. Access link (normal): https://test-actix.bluerain.io/display
1. Access link (ERR_SPDY_PROTOCOL_ERROR): https://test-actix.bluerain.io/display/w350/coding.jpeg

Link 2 is a route that returns `<fs::NamedFile>` and I need to add `proxy_http_version 1.1;` to access it. Why is Link 1 not required to do this?


This link is normal (added `proxy_http_version 1.1;`):
Https://irir.bluerain.io/display/w650/coding.jpeg

`test-actix` and `irir` reverse proxy to the same backend service.

What I mainly find strange is:
1. Why is link 1 normal?
1. Other backend web frameworks don't have this phenomenon
